core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1074,39 +1074,43 @@ class DAGScheduler(
     }
   }
 
-  /** Merge updates from a task to our local accumulator values */
+  /**
+   * Merge local values from a task into the corresponding accumulators previously registered
+   * here on the driver.
+   *
+   * Although accumulators themselves are not thread-safe, this method is called only from one
+   * thread, the one that runs the scheduling loop. This means we only handle one task
+   * completion event at a time so we don't need to worry about locking the accumulators.
+   * This still doesn't stop the caller from updating the accumulator outside the scheduler,
+   * but that's not our problem since there's nothing we can do about that.
```

Comment:
I can't imagine that users would have a legitimate use-case for updating accumulators on the driver unless they're trying to use them as a substitute for AtomicLong or something like that.